src/main/kotlin/io/github/mojira/arisa/modules/AttachmentModule.kt

```diff
@@ -19,33 +19,21 @@ class AttachmentModule(
             val functions = attachments
                 .filter { endsWithBlacklistedExtensionAdapter(it.name) }
             assertNotEmpty(functions).bind()
-            val usernames = functions.getUsernames()
-            val attachmentNames = functions.getAttachmentNames()
+            val commentInfo = functions.getCommentInfo()
             functions
                 .map { it.remove }
                 .forEach { it.invoke() }
             addComment(CommentOptions(attachmentRemovedMessage))
-            addRawRestrictedComment("Attachment Details:\nFilename: $attachmentNames\nUploader: $usernames", "helper")
+            addRawRestrictedComment("Attachments:\n$commentInfo", "helper")
         }
     }
 
-    private fun List<Attachment>.getUsernames() = this
-        .map { it.uploader!!.name }
+    private fun List<Attachment>.getCommentInfo() = this
         .run {
             when (size) {
-                1 -> get(0)
-                2 -> "${get(0)}* and *${get(1)}"
-                else -> "${subList(0, lastIndex).joinToString("*, *")}*, and *${last()}"
-            }
-        }
-
-    private fun List<Attachment>.getAttachmentNames() = this
-        .map { it.name }
-        .run {
-            when (size) {
-                1 -> get(0)
-                2 -> "${get(0)} and ${get(1)}"
-                else -> "${subList(0, lastIndex).joinToString(", ")}, and ${last()}"
+                1 -> "[~${get(0).uploader!!.name}]: ${get(0).name}"
+                else -> subList(0, lastIndex)
+                    .map { "[~${get(0).uploader!!.name}]: ${get(0).name}\n" }
             }
         }
```

Comment:
It looks like currently the return type is `Any` because `map` returns a `List`. Additionally `subList(0, lastIndex)` would exclude the last element, wouldn't it?
What do you think about the following instead?
```kotlin
    private fun List<Attachment>.getCommentInfo() =
        map { " - [~${it.uploader!!.name}]: ${it.name}" }
            .joinToString { "\n" }
```

Reply:
Looks like a better solution 👍